internal/atls/issuer/issuer.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2024 Edgeless Systems GmbH
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package issuer
+
+import (
+	"context"
+	"encoding/asn1"
+	"fmt"
+	"log/slog"
+
+	snpissuer "github.com/edgelesssys/contrast/internal/attestation/snp/issuer"
+	tdxissuer "github.com/edgelesssys/contrast/internal/attestation/tdx/issuer"
+	"github.com/edgelesssys/contrast/internal/logger"
+	"github.com/klauspost/cpuid/v2"
+)
+
+// New creates an attestation issuer for the current platform.
+func New(log *slog.Logger) (Issuer, error) {
+	cpuid.Detect()
+	switch {
+	case cpuid.CPU.Supports(cpuid.SEV_SNP):
+		return snpissuer.New(
+			logger.NewWithAttrs(logger.NewNamed(log, "issuer"), map[string]string{"tee-type": "snp"}),
+		), nil
+	case cpuid.CPU.Supports(cpuid.TDX_GUEST):
+		return tdxissuer.New(
+			logger.NewWithAttrs(logger.NewNamed(log, "issuer"), map[string]string{"tee-type": "tdx"}),
+		), nil
+	default:
+		return nil, fmt.Errorf("unsupported platform: %T", cpuid.CPU)
+	}
+}
+
+// Issuer issues an attestation document.
+type Issuer interface {
```

Comment:
The interface is also declared here: https://github.com/edgelesssys/contrast/blob/c75797e083533b282f3c19082e582e02dfd0bb82/internal/atls/atls.go#L88. I think we should only have one - here - which requires inlining the `Getter`.